test/Main.hs

```diff
@@ -1,6 +1,44 @@
 module Main ( main ) where
 
 import Test.DocTest
+import Test.Tasty
+import Test.Tasty.HUnit
+
+import Cabal2nix ( mainWithArgs )
+import System.Exit ( ExitCode (..) )
+import System.FilePath ( (</>) )
+import System.Process ( readProcessWithExitCode )
+import System.IO.Silently ( capture_ )
 
 main :: IO ()
-main = doctest [ "-isrc", "src" ]
+main = doctest [ "-isrc", "src" ] >> defaultMain tests
+
+tests :: TestTree
+tests = testGroup "Tests" [regressionTests]
+
+regressionTests :: TestTree
+regressionTests = testGroup "Regression"
+  [ testCase "#274: Generating an expression from a local file is broken since --subpath was added" $ do
+      parsec <- allCabalHashes "parsec/3.0.0/parsec.cabal"
+      out <- capture_ $ mainWithArgs [ "--sha256=dontworry", parsec]
+      out @?= unlines
+        [ "{ mkDerivation, base, bytestring, mtl, stdenv }:"
+        , "mkDerivation {"
+        , "  pname = \"parsec\";"
+        , "  version = \"3.0.0\";"
+        , "  sha256 = \"dontworry\";"
+        , "  libraryHaskellDepends = [ base bytestring mtl ];"
+        , "  homepage = \"http://www.cs.uu.nl/~daan/parsec.html\";"
+        , "  description = \"Monadic parser combinators\";"
+        , "  license = stdenv.lib.licenses.bsd3;"
+        , "}"
+        ]
+  ]
+
+
+allCabalHashes :: FilePath -> IO String
```

Comment:
This may be a bit much. The test does not require a 1,3 GByte git repository of Cabal files --- it just needs *one* Cabal file.

Also. note that this test fails when Nix runs it because there's no Nix available inside of the build sandbox. Neither does the sandbox have network access and the ability to download stuff from the Internet.